zebra/zserv.c

```diff
@@ -1031,6 +1027,10 @@ void zserv_event(struct zserv *client, enum zserv_event event)
 {
 	switch (event) {
 	case ZSERV_ACCEPT:
+		if (zsock < 0) {
```

Comment:
hmm, gosh, I sort of think the test was good in the callback: at shutdown, we don't have a clear way to cancel an existing "read" task, right?

Reply:
Nah, there's no way to cancel an existing read during shutdown. 

I moved the zsock check to this function because: In zserv_accept() func, accept() call will fail with EBADF if zsock is not an open FD. And in case some other func calls zserv_event(NULL, ZSERV_ACCEPT); in the future, the caller won't have to do an explict check for zsock.

Reply:
Let me know your thoughts. I am ok moving this check back to zserv_accept() or keeping it here.